include/ardpolyfills/BoardData.cxx

```diff
@@ -0,0 +1,52 @@
+/*
+ *  BoardData.cxx
+ *  Copyright 2020 ItJustWorkTM
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+
+
+#include <ardpolyfills/Arduino.h>
+// Arduino Math functions
+long constrain(long amt, long low, long high) { return ((amt) < (low) ? (low) : ((amt) > (high) ? (high) : (amt))); }
+
+long map(long x, long in_min, long in_max, long out_min, long out_max) { return (x - in_min) * (out_max - out_min) / (in_max - in_min) + out_min; }
+
+long min(long a, long b) { return ((a) < (b) ? (a) : (b)); }
+long max(long a, long b) { return ((a) > (b) ? (a) : (b)); }
+long sq(long x) { return (x * x); }
+
+// Arduino Characters functions
+bool isAlpha(char thisChar) { return std::isalpha(thisChar); }
+bool isAlphaNumeric(char thisChar) { return std::isalnum(thisChar); }
+
+// for some reason it gives me error i dont know why
+//bool isAscii(char thisChar) { return std::isascii(thisChar); }
```

Comment:
If it gives an error, then there's something to be fixed.
It might have to do with the fact that `std::isascii` does not exist in the C++ standard library

Reply:
I found that on the internet here is the link: https://www.ibm.com/support/knowledgecenter/ssw_ibm_i_71/rtref/isascii.htm

Reply:
Always use https://en.cppreference.com/w/